events/interactionCreate.js

```diff
@@ -5,7 +5,62 @@ const Controller = require("../util/Controller");
  * @param {import("../lib/DiscordMusicBot")} client
  * @param {import("discord.js").Interaction}interaction
  */
-module.exports = (client, interaction) => {
+module.exports = async(client, interaction) => {
+
+if (interaction.isAutocomplete()) {
+	switch (interaction.commandName) {
+	case 'play':	
+	  /**
+           * @type {import("discord.js").AutocompleteFocusedOption}
+           */
+          const focused = interaction.options.getFocused(true);
+
+          if (focused.name === "query") {
+            if (focused.value === "") return;
+            /**
+             * @type {SearchResult}
+             */
+            const result = await client.manager.search(
+              focused.value,
+              interaction.user
+            );
+
+            if (result.loadType === "TRACK_LOADED" || "SEARCH_RESULT") {
+              /**
+               * @type {Track[]}
+               */
+              const sliced = result.tracks.slice(0, 5).sort();
+
+              if (
+                focused.value.match(
+                  /(?:https:\/\/open\.spotify\.com\/|spotify:)(?:.+)?(track|playlist|artist|episode|show|album)[\/:]([A-Za-z0-9]+)/ ||
```

Comment:
Won't even work as expected. This will only check for spotify link. Hilarious

Reply:
` /^.*(youtu.be\/|list=)([^#\&\?]*).*/,` this should check youtube link